src/chia_plot.cpp

```diff
@@ -178,18 +184,40 @@ int main(int argc, char** argv)
 		"d, finaldir", "Final directory (default = <tmpdir>)", cxxopts::value<std::string>(final_dir))(
 		"p, poolkey", "Pool Public Key (48 bytes)", cxxopts::value<std::string>(pool_key_str))(
 		"f, farmerkey", "Farmer Public Key (48 bytes)", cxxopts::value<std::string>(farmer_key_str))(
+		"hwinfo", "Print information regarding underlying hardware")(
 		"help", "Print help");
 	
 	if(argc <= 1) {
 		std::cout << options.help({""}) << std::endl;
 		return 0;
 	}
 	const auto args = options.parse(argc, argv);
+
+	if(maxThreads == cinfo.logicalCpus() * 2) 
+		mPier=2;
+        else if(maxThreads == cinfo.logicalCpus() * 4)
+		mPier=4;
+	maxCores=cinfo.cores() * mPier;
```

Comment:
what's the logic behind `mPier`? I don't get it..

Reply:
The thread is running on 1 core at a time and the ASM calls will give info about the running CPU. There is no easy cross platform way to know all the CPUs on the system. So we take advantage of the std::thread:hardware_concurrency call which will report the total logicalprocessors on the system. Having that information and knowing the setup of one physical CPU we can deduct setups of dual and quad processor systems taking in mind that the setup is symmetric.  mPier is the multiplier of the single CPU cores according to the calculation above.

Reply:
so `mPier` == number of CPUs?

Reply:
Exactly.